### PR TITLE
Add Terraform R2 Backend 

### PR DESCRIPTION
**Pull request by SomeBadCoding**
Add Terraform R2 Backend Example

### Summary

Add Terraform R2 Backend Example. Include process for creating R2 Bucket and API Tokens. 

```
content/terraform/_partials/_create-r2-buckets.md
content/terraform/advanced-topics/remote-backend.md
```

### CODE REVIEW

**Comment by ranbel** [comment]
Hi @angelampcosta and @dcpena, would you mind reviewing the Logpull and R2 content?

File: content/logs/logpull/enabling-log-retention/index.md [hunk 1]
```diff
@@ -23,55 +23,32 @@ To make a `POST` call, you must have zone-scoped `edit` permissions, such as Sup
...
 
-### Check whether log retention is turned on:
+### Check log retention status:
 
-```sh
-$ curl -s -H "X-Auth-Email: <EMAIL>" -H "X-Auth-Key: <API_KEY>" -X GET "https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/logs/control/retention/flag" | jq .
-```
+In the linux examples below we use optional [jq](/logs/tutorials/parsing-json-log-data/) tool to help parse the response.
 
-#### Response:
```
Comment (soheiokamoto) [inline content/logs/logpull/enabling-log-retention/index.md:32]: Is it intentional to remove the response completely here? It's not in `_check-log-retention.md` partial.

File: content/logs/logpull/enabling-log-retention/index.md [hunk 2]
```diff
@@ -23,55 +23,32 @@ To make a `POST` call, you must have zone-scoped `edit` permissions, such as Sup
...
-
-```bash
-curl -s -H "X-Auth-Email: <EMAIL>" -H "X-Auth-Key: <API_KEY>" -X POST "https://api.cloudflare.com/client/v4/zones/<ZONE_ID>/logs/control/retention/flag" -d'{"flag":true}' | jq .
-```
+{{<render file="_enable-log-retention.md">}}
 
-On Windows in Command Prompt:
+#### Enabled Response:
 
+```json
```
Comment (soheiokamoto) [inline content/logs/logpull/enabling-log-retention/index.md:42]: Are we just showing the `result` part in the response?
`result` is extracted using `jq` only in `Linux` example, but not for the others.

Same for `Disable Response`.

File: content/logs/_partials/_check-log-retention.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  render: never
+  list: never
+---
+
+{{<tabs labels="Linux | CMD | Powershell">}}
+{{<tab label="linux" default="true">}}
+
+```bash
+ curl --silent --request GET \
+  --url https://api.cloudflare.com/client/v4/zones/<ZONEID>/logs/control/retention/flag \
```
Comment (soheiokamoto) [inline content/logs/_partials/_check-log-retention.md]: You might want to update the variable part to keep it consistent with the guidance: https://developers.cloudflare.com/style-guide/api-content-strategy/guidelines-for-curl-commands/
```suggestion
  --url https://api.cloudflare.com/client/v4/zones/{zone_id}/logs/control/retention/flag \
```

Same for other endpoints.

File: content/logs/logpull/enabling-log-retention/index.md (hunk 1 above)
Comment (SomeBadCoding) [reply content/logs/logpull/enabling-log-retention/index.md:32]: Yes, I've added some additional wording for the quick answer, and link to the response example

File: content/logs/logpull/enabling-log-retention/index.md (hunk 2 above)
Comment (SomeBadCoding) [reply content/logs/logpull/enabling-log-retention/index.md:42]: Yes, moved the sentence which mentions this into the note. Should keep the CMD versions generic

File: content/logs/_partials/_check-log-retention.md (hunk 3 above)
Comment (SomeBadCoding) [reply content/logs/_partials/_check-log-retention.md]: Adjusted to purposed version, and removed some extra spaces.

File: content/terraform/_partials/_create-r2-buckets.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+```sh
+$ wrangler r2 bucket create your-tfstate-bucket-name
+```
+
+{{</tab>}}
+{{<tab label="api">}}
+
+```bash
+ curl --request POST \
+  --url https://api.cloudflare.com/client/v4/accounts/<account_id>/r2/buckets \
```
Comment (soheiokamoto) [inline content/terraform/_partials/_create-r2-buckets.md]: nit, to make similar to https://developers.cloudflare.com/style-guide/api-content-strategy/guidelines-for-curl-commands/
```suggestion
  --url https://api.cloudflare.com/client/v4/accounts/{account_id}/r2/buckets \
```

File: content/terraform/_partials/_create-r2-buckets.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  --header 'Authorization: Bearer <token>' \
```
Comment (soheiokamoto) [inline content/terraform/_partials/_create-r2-buckets.md]: nit, per https://developers.cloudflare.com/style-guide/api-content-strategy/guidelines-for-curl-commands/
```suggestion
  --header "Authorization: Bearer <API_TOKEN>" \
```
Also, to be consistent, you might want to use the `"` for all others.

**Review by soheiokamoto — APPROVED** [review]

File: content/terraform/_partials/_create-r2-buckets.md (hunk 5 above)
Comment (SomeBadCoding) [reply content/terraform/_partials/_create-r2-buckets.md]: Make sense, I've made the corresponding adjustments.

**Review by angelampcosta — APPROVED** [review]

**Comment by soheiokamoto** [comment]
Can we have this merged?
Waiting for this to be merged, then will update https://github.com/cloudflare/cloudflare-docs/pull/16352

**Comment by angelampcosta** [comment]
Closed by https://github.com/cloudflare/cloudflare-docs/pull/16790

**Pull request closed without merging**